support/documentation/architecture/decisions/0004-project-structure.md

```diff
@@ -0,0 +1,39 @@
+# Project structure
+
+- Status: accepted
+- Date: 2020-05-11
+
+## Context and Problem Statement
+
+We need a consistent way of structuring code in the project. How should we structure our code?
+
+## Decision Outcome
+
+Our project should consist of multiple modules where each of the modules will have only one clearly defined purpose. Our modules will consist of three separate modules (where applicable, since not all
+the modules require such a separation), the `api` module will contain public interfaces, classes for method parameters and results of those interfaces, `core` module containing the implementation that
+depends on the api module and `spring-boot` module that contains Spring Boot's autoconfiguration for specified module and has dependencies on both `api` and `core` modules. Services in `core` module
+implementing the interface from the `api` module should have a `Default` prefix. `spring-boot` module sole responsibility should only be providing the configuration for `core` module (the
```

Comment:
Not sure this is about `Project structure`. It seems more like `Coding conventions` or something like that.

Reply:
Yes I was not sure about the title too, but it's not actually coding convention either, if we can come up with something better we can change it.

Reply:
`Naming conventions` is probably more accurate here.

Reply:
Don't know for me the closest currently is project structure.

Reply:
We can keep it here for now until we write a new ADR for coding conventions